### PR TITLE
Add forcePathStyle option

### PR DESCRIPTION
**Pull request by andrzejressel**
`forcePathStyle` is required when connecting to MinIO with default configuration using hostname.

### CODE REVIEW

File: zio-s3/src/main/scala/zio/s3/Live.scala [hunk 1]
```diff
@@ -234,7 +234,8 @@ object Live {
   def connect[R](
     region: S3Region,
     provider: RIO[R with Scope, AwsCredentialsProvider],
-    uriEndpoint: Option[URI]
+    uriEndpoint: Option[URI],
+    forcePathStyle: Option[Boolean]
```
Comment (regis-leray) [inline zio-s3/src/main/scala/zio/s3/Live.scala]: use a boolean is enough :) with a default value  to `false` please double check if it is the default value in sts library

File: zio-s3/src/main/scala/zio/s3/settings.scala [hunk 2]
```diff
@@ -40,10 +40,14 @@ object S3Region { self =>
     new S3Region(Region.of(r)) {}
 }
 
-final case class S3Settings(s3Region: S3Region, credentials: AwsCredentials)
+final case class S3Settings(s3Region: S3Region, credentials: AwsCredentials, forcePathStyle: Option[Boolean])
```
Comment (regis-leray) [inline zio-s3/src/main/scala/zio/s3/settings.scala]: i dont think we need to add this parameter to the settings, see my comment ☝️

File: zio-s3/src/main/scala/zio/s3/package.scala [hunk 3]
```diff
@@ -32,25 +32,35 @@ import java.util.concurrent.CompletableFuture
...
     ZLayer.scoped[R](
       ZIO
         .fromEither(S3Region.from(region))
-        .flatMap(Live.connect[R](_, provider, uriEndpoint))
+        .flatMap(Live.connect[R](_, provider, uriEndpoint, forcePathStyle))
     )
 
   val live: ZLayer[S3Settings, ConnectionError, S3] = ZLayer.scoped(
-    ZIO.serviceWithZIO[S3Settings](s => Live.connect(s.s3Region, const(s.credentials), None))
+    ZIO.serviceWithZIO[S3Settings](s => Live.connect(s.s3Region, const(s.credentials), None, s.forcePathStyle))
```
Comment (regis-leray) [inline zio-s3/src/main/scala/zio/s3/package.scala]: please remove from here, if people want to use forcePath they will use the liveZIO or live function

File: zio-s3/src/main/scala/zio/s3/Live.scala (hunk 1 above)
Comment (andrzejressel) [reply zio-s3/src/main/scala/zio/s3/Live.scala]: Unfortunately false value is not the same as no value:

```java
// DefaultS3BaseClientBuilder.java

        if (serviceConfigBuilder.pathStyleAccessEnabled() != null) {
            Validate.validState(
                    clientContextParams.get(S3ClientContextParams.FORCE_PATH_STYLE) == null,
                    "ForcePathStyle has been configured on both S3Configuration and the client/global level. Please limit ForcePathStyle configuration to one location.");
        } else {
            serviceConfigBuilder.pathStyleAccessEnabled(clientContextParams.get(S3ClientContextParams.FORCE_PATH_STYLE));
        }
```

File: zio-s3/src/main/scala/zio/s3/settings.scala (hunk 2 above)
Comment (andrzejressel) [reply zio-s3/src/main/scala/zio/s3/settings.scala]: Fixed

File: zio-s3/src/main/scala/zio/s3/package.scala (hunk 3 above)
Comment (andrzejressel) [reply zio-s3/src/main/scala/zio/s3/package.scala]: Fixed

**Comment by andrzejressel** [comment]
Due to https://github.com/zio/zio-s3/pull/476#discussion_r1578487496 I've restored Option

**Review by regis-leray — APPROVED** [review]

**Pull request merged**